### PR TITLE
xattrs / extended stat: improve exception handling (master)

### PR DESCRIPTION
**Pull request by KN4CK3R**
forward port of #6990

The different quotes in both branches made some trouble but should be fine now.

### CODE REVIEW

File: src/borg/testsuite/archiver.py [hunk 1]
```diff
@@ -1560,7 +1560,7 @@ def patched_fchown(fd, uid, gid):
             assert xattr.getxattr(b"input/file", b"security.capability") == capabilities
 
     @pytest.mark.skipif(
-        not xattr.XATTR_FAKEROOT, reason="xattr not supported on this system or on this version of" "fakeroot"
+        not xattr.XATTR_FAKEROOT, reason="xattr not supported on this system or on this version of " "fakeroot"
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py:1563]: hmm, if that's what black made out of this, guess we better merge the 2 strings into 1.

see also the other 2 places with same string.

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]

File: src/borg/testsuite/archiver.py (hunk 1 above)
Comment (ThomasWaldmann) [reply src/borg/testsuite/archiver.py:1563]: @KN4CK3R did you see my feedback?

**Review by ThomasWaldmann — APPROVED** [review]
fixed the cosmetics myself so i can merge this soon.

**Comment by KN4CK3R** [comment]
Sorry, was not a home.

**Pull request merged**